### ISSUE TITLE
Feature Request - Add option to config file to change colours there

### ISSUE THREAD

**Issue by thommo17** [ScottSturdivant/rpi_metar#22]
Hi Scott,

I've been toying with the idea of being able to change the METAR colours via the config file.

Something like this I guess

`class FlightCategory(Enum, cfg):
    VFR = cfg.get('settings', 'VFR', fallback=GREEN)
    IFR = cfg.get('settings', 'IFR', fallback=YELLOW)
    MVFR = cfg.get('settings', 'MVFR', fallback=BLUE)
    LIFR = cfg.get('settings', 'LIFR', fallback=RED)
    UNKNOWN = BLACK
    OFF = BLACK
    MISSING = MAGENTA`

Not sure how to make the changes to the wx.py file to get it to read the conf file though

**Comment by ScottSturdivant** [comment]
This is an interesting idea! Let me ask you this: if you're going through the trouble to change the colors via configuration, would you be satisfied with the preset RGB/GRB values for those colors, or might you want to set those at the same time?

```
[settings]
VFR = GREEN # One option, uses default
IFR = (254, 1, 0)  # A custom green GRB value.
```

**Comment by thommo17** [comment]
That's a really good idea!

Opens up the user to a lot more possibilities.

Also helps with testing the next thing I was thinking of implementing which is night mode colours.

I was going to try and have a different set of colours after a set time or maybe even have it know when sunrise/sunset is and change the colours accordingly.

The colours were just going to be the same RGB but added a bit more darkness to them instead of being like 255, 0, 0 etc

**Comment by ScottSturdivant** [comment]
Sounds good, I'll look at giving this a shot over the weekend.

**Comment by ScottSturdivant** [comment]
After playing with this for a bit, I've got a question for you. Do you intend to actually change the association of a flight category to a color (e.g. change VFR from GREEN to MAGENTA), or are you wishing to change how a color itself is represented (e.g. (255,0,0) to (245,0,1), or possibly all of the above?

**Comment by thommo17** [comment]
I'd say both would be good.

Could do just the colour (GREEN) though and in the LED file I could add as many colour designations as I can think of.

Being able to do RGB numbers would make it very nice though for testing.

**Comment by ScottSturdivant** [comment]
There's an attempt at this on the https://github.com/ScottSturdivant/rpi_metar/tree/custom_colors branch. Would you mind taking a look and giving me your feedback?

The README.md is updated with information about how to do this in the `colors` and `flight_categories` sections. I hope the information there is clear enough!

**Comment by thommo17** [comment]
Awesome!

I'll give it a shot today and see how it works.

Once again thanks for your hard work on this code

**Comment by thommo17** [comment]
All set up and running beautifully. Looks like it was a bit of work to get it all organised.

I tried to see what a few different colours would look like but some just don't work well. For instance the NAVY_BLUE colour will just be blank but I think that is because it is so close to black the lights can't show it.

I'm going to keep testing things for a bit but I think you've smashed it out of the park once again

**Comment by ScottSturdivant** [comment]
Closed by #24
